vaadin-grid-flow-demo/src/main/java/com/vaadin/flow/component/grid/demo/GridDemo.java

```diff
@@ -2059,6 +2073,322 @@ private void createNotBufferedDynamicEditor() {
+                index = targetItems.indexOf(event.getDropTargetRow().get())
+                        + (event.getDropLocation() == DropLocation.BELOW ? 1
+                                : 0);
+            }
+
+            targetItems.addAll(index, draggedItems);
+            targetGrid.setItems(targetItems);
+
+            draggedItems = null;
+        };
+
+        PersonService personService = new PersonService();
+
+        Grid<Person> grid = new Grid<>(Person.class);
+        grid.setItems(personService.fetch(0, 5));
+        grid.setSelectionMode(Grid.SelectionMode.MULTI);
+        grid.addDropListener(dropListener);
+        grid.addDragStartListener(dragStartListener);
+        grid.setRowsDraggable(true);
+        grid.setDropMode(DropMode.BETWEEN);
+        grid.setColumns("firstName", "lastName");
+
+        Grid<Person> grid2 = new Grid<>(Person.class);
+        grid2.setItems(personService.fetch(5, 1));
+        grid2.setSelectionMode(Grid.SelectionMode.MULTI);
+        grid2.addDropListener(dropListener);
+        grid2.addDragStartListener(dragStartListener);
+        grid2.setRowsDraggable(true);
+        grid2.setDropMode(DropMode.BETWEEN);
+        grid2.setColumns("firstName", "lastName");
+
+        // end-source-example
+
+        HorizontalLayout hl = new HorizontalLayout(grid, grid2);
+
+        addCard("Drag and Drop", "Drag Rows Between Grids", hl);
+    }
+
+    private void createDropLocations() {
+
+        // begin-source-example
+        // source-example-heading: Drop Location
+        PersonService personService = new PersonService();
+
+        Grid<Person> grid = new Grid<>(Person.class);
+        grid.setItems(personService.fetch(0, 50));
+
+        grid.addDragStartListener(
+                event -> draggedItems = event.getDraggedItems());
+        grid.setColumns("firstName", "lastName", "phoneNumber");
+        grid.setRowsDraggable(true);
+
+        TreeData<Person> td = new TreeData<>();
+        td.addItems(null, personService.fetch(51, 2));
+
+        TreeGrid<Person> treeGrid = new TreeGrid<>();
+        treeGrid.setDataProvider(new TreeDataProvider<Person>(td));
+        /*
+         * This enables dropping in between or on top of the existing grid rows.
+         */
+        treeGrid.setDropMode(DropMode.ON_TOP_OR_BETWEEN);
+        treeGrid.addHierarchyColumn(Person::getfirstName)
+                .setHeader("firstName");
+        treeGrid.addColumn(Person::getLastName).setHeader("lastName");
+        treeGrid.addColumn(Person::getPhoneNumber).setHeader("phoneNumber");
+        treeGrid.addDropListener(event -> {
+            // Remove the items from the source grid
+            ListDataProvider<Person> sourceDataProvider = (ListDataProvider<Person>) grid
+                    .getDataProvider();
+            List<Person> sourceItems = new ArrayList<Person>(
+                    sourceDataProvider.getItems());
+            sourceItems.remove(draggedItems.get(0));
```

Comment:
This breaks when dragging something else than rows (no draggedItems)

Reply:
Disabled drop mode by default